Dima/netflix-roulette/src/Pages/FilmPage/FilmPage.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import Header from '../../Components/Header/Header'
+import Button from '../../Shared/Buttons/Button'
+import FilmInfo from '../../Components/FilmInformation/FilmInformation'
+import AdditionalPanel from '../../Components/AdditionalPanel/AdditionalPanel'
+import Films from '../../Components/Films/Films'
+import Footer from '../../Components/Footer/Footer'
+// import IFilmProps from '../../interfaces/IFIlmProps'
+import { Link, RouteComponentProps } from 'react-router-dom'
+import { CurrentFilmRequested, FilmsRequested } from 'redux/Actions/requestActions'
+import { connect, ConnectedProps } from 'react-redux'
+import IFilm from 'interfaces/IFilm'
+import { parse } from 'query-string'
+
+interface IRootState {
+  films: any, //and here too((((
+  currentFilm: IFilm,
+  id: string
+};
+const mapStateToProps = (state: IRootState) => ({
+  films: state.films.films,
+  currentFilm: state.films.currentFilm,
+  id: state.id
+});
+const mapDispatchToProps = {
+  CurrentFilmRequested,
+  FilmsRequested,
+}
+
+const connector = connect(mapStateToProps, mapDispatchToProps)
+
+type PropsFromRouteAndRedux = ConnectedProps<typeof connector> & RouteComponentProps<{ id: string }>
+
+
+class FilmPage extends React.Component<PropsFromRouteAndRedux> {
+  componentDidMount() {
+    this.props.CurrentFilmRequested(this.props.match.params.id);
+  };
+  
+  componentDidUpdate(prevProps: PropsFromRouteAndRedux) {
```

Comment:
not valid. 
You have to write an if statement here.  componentDidUpdate will be executed every time on every prop change, so you have to write a check!